src/main/java/com/mraof/minestuck/client/gui/playerStats/EcheladderScreen.java

```diff
@@ -332,27 +321,66 @@ public boolean mouseClicked(double xcor, double ycor, int mouseButton)
 		{
 			if(ycor >= yOffset + 35 && ycor < yOffset + 42)
 			{
-				scrollIndex = Mth.clamp(scrollIndex + RUNG_Y, 0, maxScroll);
-				isScrolling = true;
+				scroll = maxScroll;
 				return true;
 			} else if(ycor >= yOffset + 185 && ycor < yOffset + 192)
 			{
-				scrollIndex = Mth.clamp(scrollIndex - RUNG_Y, 0, maxScroll);
-				isScrolling = true;
+				scroll = 0;
 				return true;
 			}
 		}
+		
 		return super.mouseClicked(xcor, ycor, mouseButton);
 	}
 	
-	@Override
-	public boolean mouseReleased(double mouseX, double mouseY, int mouseButton)
+	private final class RungBar extends AbstractWidget
 	{
-		if(isScrolling)
+		private final int initY;
+		private final int rung;
+		
+		public RungBar(int initX, int initY, int width, int height, Component message, Optional<String> tooltip, int rung)
+		{
+			super(initX, initY, width, height, message);
+			this.initY = initY;
+			this.rung = rung;
+			tooltip.ifPresent(string -> setTooltip(Tooltip.create(Component.translatable(string))));
+		}
+		
+		@Override
+		public void renderWidget(GuiGraphics guiGraphics, int mouseX, int mouseY, float partialTick)
+		{
+			int x = getX();
+			int y = getY();
```

Comment:
The y variable is well-used, but I think x could be used more too, basically anywhere where xOffset is used in this function.